x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/MlIndexAndAlias.java

```diff
@@ -390,4 +393,11 @@ public static String latestIndex(String[] concreteIndices) {
             ? concreteIndices[0]
             : Arrays.stream(concreteIndices).max(MlIndexAndAlias.INDEX_NAME_COMPARATOR).get();
     }
+
+    /**
+     * True if the version is read *and* write compatible not just read only compatible
+     */
+    public static boolean indexIsReadWriteCompatibleInV9(IndexVersion version) {
+        return version.onOrAfter(IndexVersions.V_8_0_0);
```

Comment:
I would call it something like `indexIsInCurrentFormat` so it is also valid on main. In 9.0.0 the same code should then check for `IndexVersions.V_9_0_0`